code/game/machinery/firealarm.dm

```diff
@@ -348,6 +348,25 @@
 	else
 		set_light(l_power = 0)
 
+// Allows users to examine the state of the thermal sensor
+/obj/machinery/firealarm/examine(mob/user)
+	. = ..()
+	. += "A light on the side indicates the thermal sensor is [detecting ? "enabled" : "disabled"]."
+
+// Allows Silicons to disable thermal sensor
+/obj/machinery/firealarm/BorgCtrlClick(mob/living/silicon/robot/user)
+	if(get_dist(src,user) <= user.interaction_range)
+		AICtrlClick(user)
+	else
+		..()
```

Comment:
```suggestion
/obj/machinery/firealarm/BorgCtrlClick(mob/living/silicon/robot/user)
	if(get_dist(src,user) <= user.interaction_range)
		AICtrlClick(user)
		return
	return ..()
```
* Preserve parent returns unless you have specific reasons not to.
* Make use and abuse of early returns to diminish horizontal drift.

Reply:
Huh gotcha shall keep that in mind for the future